elide-async/src/main/java/com/yahoo/elide/async/service/AsyncExecutorService.java

```diff
@@ -85,17 +90,50 @@ public synchronized static AsyncExecutorService getInstance() {
      * Execute Query asynchronously.
      * @param queryObj Query Object
      * @param user User
+     * @param apiVersion API Version
      */
     public void executeQuery(AsyncQuery queryObj, User user, String apiVersion) {
+
         QueryRunner runner = runners.get(apiVersion);
         if (runner == null) {
             throw new InvalidOperationException("Invalid API Version");
         }
-
         AsyncQueryThread queryWorker = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, apiVersion);
 
-        AsyncQueryInterruptThread queryInterruptWorker = new AsyncQueryInterruptThread(elide,
-               executor.submit(queryWorker), queryObj, new Date(), maxRunTime, asyncQueryDao);
-        interruptor.execute(queryInterruptWorker);
+        Future<AsyncQueryResult> task = executor.submit(queryWorker);
+
+        try {
+            queryObj.setStatus(QueryStatus.PROCESSING);
+            AsyncQueryResult queryResultObj = task.get(queryObj.getAsyncAfterSeconds(), TimeUnit.SECONDS);
+            queryObj.setResult(queryResultObj);
```

Comment:
It looked like your unit tests were verifying null was returned when a runtime exception was thrown.  However, we don't handle null here.